insights/parsers/tests/test_initscript.py

```diff
@@ -0,0 +1,152 @@
+from insights.tests import context_wrap
+
+
+NOTINITSCRIPT_SCRIPT = "etc/rc.d/init.d/script_notinitscript"
+NOTINITSCRIPT_CONTENT = """
+#!/bin/bash
+
+echo script_notinitscript
+""".strip()
+
+CHKCONFIG_SCRIPT = "etc/rc.d/init.d/script_chkconfig"
+CHKCONFIG_CONTENT = """
+#!/bin/sh
+#
+# rhnsd:        Starts the Spacewalk Daemon
+#
+# chkconfig: 345 97 03
+# description:  This is a daemon which handles the task of connecting \
+#               periodically to the Spacewalk servers to \
+#               check for updates, notifications and perform system \
+#               monitoring tasks according to the service level that \
+#               this server is subscribed for
+#
+# processname: rhnsd
+# pidfile: /var/run/rhnsd.pid
+#
+
+echo script_chkconfig
+""".strip()
+
+LSB_SCRIPT = "etc/rc.d/init.d/script_lsb"
+LSB_CONTENT = """
+#!/bin/sh
+### BEGIN INIT INFO
+# Provides: rhnsd
+# Required-Start: $local_fs $network $remote_fs $named $time
+# Required-Stop: $local_fs $network $remote_fs $named
+# Default-Start: 2 3 4 5
+# Default-Stop: 0 1 6
+# Short-Description: Starts the Spacewalk Daemon
+# Description: This is a daemon which handles the task of connecting
+#               periodically to the Spacewalk servers to
+#               check for updates, notifications and perform system
+#               monitoring tasks according to the service level that
+#               this server is subscribed for.
+### END INIT INFO
+
+echo script_lsb
+""".strip()
+
+CHKCONFIG_LSB_SCRIPT = "etc/rc.d/init.d/script_chkconfig+lsb"
+CHKCONFIG_LSB_CONTENT = """
+#!/bin/sh
+#
+# rhnsd:        Starts the Spacewalk Daemon
+#
+# chkconfig: 345 97 03
+# description:  This is a daemon which handles the task of connecting \
+#               periodically to the Spacewalk servers to \
+#               check for updates, notifications and perform system \
+#               monitoring tasks according to the service level that \
+#               this server is subscribed for
+#
+# processname: rhnsd
+# pidfile: /var/run/rhnsd.pid
+#
+
+### BEGIN INIT INFO
+# Provides: rhnsd
+# Required-Start: $local_fs $network $remote_fs $named $time
+# Required-Stop: $local_fs $network $remote_fs $named
+# Default-Start: 2 3 4 5
+# Default-Stop: 0 1 6
+# Short-Description: Starts the Spacewalk Daemon
+# Description: This is a daemon which handles the task of connecting
+#               periodically to the Spacewalk servers to
+#               check for updates, notifications and perform system
+#               monitoring tasks according to the service level that
+#               this server is subscribed for.
+### END INIT INFO
+
+echo script_chkconfig+lsb
+""".strip()
+
+EMPTY_SCRIPT = "etc/rc.d/init.d/script_empty"
+EMPTY_CONTENT = """
+""".strip()
+
+HINTSONLY_SCRIPT = "etc/rc.d/init.d/script_hintsonly"
+HINTSONLY_CONTENT = """
+#!/bin/sh
+#
+
+case "$1" in
+start)
+        ;;
+stop)
+        ;;
+esac
+""".strip()
+
+
+class TestInitScript(unittest.TestCase):
```

Comment:
Please use pytest instead of unittest for your test cases.  All that is necessary to do that is to make all of the class methods functions, remove the class, and make all of the `self.assert...` statements `assert`.  For instance `test_initscript2` becomes:
```
def test_initscript2():
    context = context_wrap(CHKCONFIG_CONTENT)
    context.path = CHKCONFIG_SCRIPT

    r = InitScript(context)
    assert r.file_name == "script_chkconfig"
```